zebra-state/src/request.rs

```diff
@@ -53,6 +53,23 @@ impl HashOrHeight {
         }
     }
 
+    /// Unwrap the inner hash or attempt to retrieve the hash for a given
+    /// height if one exists.
+    ///
+    /// # Consensus
+    ///
+    /// In the non-finalized state, a height can have multiple valid hashes.
+    /// We typically use the hash that is currently on the best chain.
+    pub fn hash_or_else<F>(self, op: F) -> Option<block::Hash>
```

Comment:
Could this be removed if it's unused?

Reply:
We already have `height_or_else()`, and I think we'll end up using `hash_or_else()` some time pretty soon.

I almost used it in this PR, but `Fn` generics can't accept async closures.